### ISSUE TITLE
Mini batch size.

### ISSUE THREAD

**Issue by dream3dllc** [Tai-Hsien/MeshSegNet#46]
Testing with smaller minibatch is giving me completely different results. Did you ever try to use anything smaller than 10 for the batch size. Also another question i have is does it really matter if i have flip meshes or not.

**Comment by Tai-Hsien** [comment]
When I predict, the batch size is actually 1 and it works fine.
For flip, make sure you also flip your label (not just flip mesh).
The sample containing 9k faces requires more GPU memory for sure.

**Comment by dream3dllc** [comment]
Hi Tai,
Sorry i meant to say Training not testing. Yes Testing is done with an single batch. But for you current example you use an batch of 10 for the training/validation. So my question was that when i do the training/validation with an batch of 2 because that is only number that can fit on my GPU. The val_dsc stays around 0.16 to 0.24  with 200 epochs. Also another question i have for you is that in your orginal example you are using 6000 for the patch size, how did you end up getting it to train off gum data in the case where your model had more than 6000 triangles and your study in that case does not include any gum. Would that not cause issues later on when you try to Test as those faces would not get predicted properly.

**Comment by Tai-Hsien** [comment]
There are BatchNorms in the network, so I will say batch of 2 will cause the problem.

You can set 5500 for all tooth cells and 500 for gum cells. If your number of tooth cells > 5500, then just randomly select up to 5500. It's the flexible setting. No idea answer, as long as it works for you. We did include gum cells (with low percentage) in the training data.

For myself, I usually trained the model with 9000 cells and my dataset was downsampled to 10,000 cells. It works well for my application.

**Comment by dream3dllc** [comment]
I am so dumb lol. You are right BatchNorms is part of the study. I totally forgot about that. I will retrain with bigger batches and see what i get. I will let you know. Also regarding the 9k cells. That seems to work much better. I am doing something very similar to your. I downsampled all my mesh to 10k and they are label for the training. So i will continue to train with 9k that way i can guarantee to always have some gum information.
